test/core/event_engine/cf/cf_engine_test.cc

```diff
@@ -80,6 +82,211 @@ TEST(CFEventEngineTest, TestConnectionCancelled) {
+
+        resolve_signal.Notify();
+      },
+      "1.2.3.4.nip.io:80", "");
+
+  resolve_signal.WaitForNotification();
+}
+
+TEST(CFEventEngineTest, TestResolveIPv6Remote) {
+  grpc_core::Notification resolve_signal;
+
+  auto cf_engine = std::make_shared<CFEventEngine>();
+  auto dns_resolver = cf_engine->GetDNSResolver({});
+
+  dns_resolver.value()->LookupHostname(
+      [&resolve_signal](auto result) {
+        EXPECT_TRUE(result.status().ok());
+        EXPECT_THAT(
+            ResolvedAddressesToStrings(result.value()),
+            testing::UnorderedElementsAre("[2607:f8b0:400a:801::1002]:80"));
+
+        resolve_signal.Notify();
+      },
+      "2607-f8b0-400a-801--1002.sslip.io.", "80");
+
+  resolve_signal.WaitForNotification();
+}
+
+TEST(CFEventEngineTest, TestResolveIPv4Literal) {
+  grpc_core::Notification resolve_signal;
+
+  auto cf_engine = std::make_shared<CFEventEngine>();
+  auto dns_resolver = cf_engine->GetDNSResolver({});
+
+  dns_resolver.value()->LookupHostname(
+      [&resolve_signal](auto result) {
+        EXPECT_TRUE(result.status().ok());
+        EXPECT_THAT(ResolvedAddressesToStrings(result.value()),
+                    testing::UnorderedElementsAre("1.2.3.4:443"));
+
+        resolve_signal.Notify();
+      },
+      "1.2.3.4", "https");
+
+  resolve_signal.WaitForNotification();
+}
+
+TEST(CFEventEngineTest, TestResolveIPv6Literal) {
+  grpc_core::Notification resolve_signal;
+
+  auto cf_engine = std::make_shared<CFEventEngine>();
+  auto dns_resolver = cf_engine->GetDNSResolver({});
+
+  dns_resolver.value()->LookupHostname(
+      [&resolve_signal](auto result) {
+        EXPECT_TRUE(result.status().ok());
+        EXPECT_THAT(
+            ResolvedAddressesToStrings(result.value()),
+            testing::UnorderedElementsAre("[2607:f8b0:400a:801::1002]:443"));
+
+        resolve_signal.Notify();
+      },
+      "[2607:f8b0:400a:801::1002]", "443");
+
+  resolve_signal.WaitForNotification();
+}
+
+TEST(CFEventEngineTest, TestResolveNoRecord) {
+  grpc_core::Notification resolve_signal;
+  auto cf_engine = std::make_shared<CFEventEngine>();
+  auto dns_resolver = std::move(cf_engine->GetDNSResolver({})).value();
+
+  dns_resolver->LookupHostname(
+      [&resolve_signal](auto result) {
+        EXPECT_EQ(result.status().code(), absl::StatusCode::kNotFound);
+
+        resolve_signal.Notify();
+      },
+      "nonexisting-target.dns-test.event-engine.", "443");
+
+  resolve_signal.WaitForNotification();
+}
+
+TEST(CFEventEngineTest, TestResolveCanceled) {
+  grpc_core::Notification resolve_signal;
+  auto cf_engine = std::make_shared<CFEventEngine>();
+  auto dns_resolver = std::move(cf_engine->GetDNSResolver({})).value();
+
+  dns_resolver->LookupHostname(
+      [&resolve_signal](auto result) {
+        EXPECT_EQ(result.status().code(), absl::StatusCode::kCancelled);
+
+        resolve_signal.Notify();
+      },
+      "dont-care-since-wont-be-resolved.test.com", "443");
```

Comment:
This can race, right? Are you not seeing any failures?

Reply:
Technically yes, but very unlikely as it needs to connect to the server (dnssd), send a request and wait for at least two responses (ipv4 and v6) for the lookup to complete.
I didn't see it failed in my tests.

Reply:
I know you can mark the entire bazel test target as flaky, but not sure if you can do that for an individual test.

Reply:
It does fails with --runs_per_test=1000.
I"m only verifying the code if status is not ok.

Reply:
That's an improvement. I still expect that, given time, this test will also see Unknown and NotFound errors. Marking it flaky is probably sufficient for that as well.

Reply:
I changed it to localtest.me as well, so there should be no NotFound errors. For Unknown errors it can happen to other tests as well and I guess we'll need to investigate.